src/xtgeo/io/tsurf/_tsurf_reader.py

```diff
@@ -0,0 +1,799 @@
+from pathlib import Path
+from typing import Any, Generator, Optional, Union
+
+import numpy as np
+
+from xtgeo.common.types import FileLike
+
+
+class ValidatorCoordSys:
+    """
+    Validator for the coordinate system in the TSurf file format.
+    For each keyword there are multiple possible values.
+    The reader does not (yet) recognise and handle all values
+    that are valid in the TSurf format.
+
+    Note that for some of the keywords there is a short-list of allowed values.
+    The reader will issue an error when an invalid value is encountered.
+    For other keywords the reader will issue a warning
+    when an "uncommon" value is used.
+    """
+
+    # axis_names and axis_units: the whole tuple is specified, instead of
+    # single values that can be used for each position in the tuples.
+    # This is to ensure that physically meaningful relations are used.
+    # For example, it makes no sense to let two elements in axis_names be equal.
+    # Or to use two different axis_units laterally.
+
+    axis_names = [("X", "Y", "Z")]
+    """XYZ are the most common axis names"""
+
+    axis_units = [("m", "m", "m"), ("ft", "ft", "ft")]
+    """meters is the most common unit"""
+
+    z_positives = ["Depth", "Elevation"]
+    """
+    ZPOSITIVE = 'Depth': Z is increasing downwards
+    ZPOSITIVE = 'Elevation': Z is increasing upwards
+    """
+
+    @classmethod
+    def validate_coord_sys_data(
+        cls, coord_sys_data: dict[str, Any], filepath_errmsg: str
+    ) -> None:
+        """Validate the coordinate system data.
+
+        Args:
+            coord_sys_data: Dictionary with coordinate system data
+            filepath_errmsg: Error context for meaningful error messages
+
+        Raises:
+            ValueError: If coordinate system data is invalid
+        """
+
+        required_fields = ["name", "axis_name", "axis_unit", "zpositive"]
+        missing_fields = [f for f in required_fields if f not in coord_sys_data]
+        if missing_fields:
+            raise ValueError(
+                f"\nIn file {filepath_errmsg}:\n"
+                f"Coordinate system section missing fields: {missing_fields}"
+            )
+
+        # Validate coordinate system data
+        cls._validate_axis_names(tuple(coord_sys_data["axis_name"]), filepath_errmsg)
+
+        cls._validate_axis_units(tuple(coord_sys_data["axis_unit"]), filepath_errmsg)
+
+        cls._validate_zpositive(coord_sys_data["zpositive"], filepath_errmsg)
+
+    @classmethod
+    def _validate_axis_names(
+        cls, axis_names: tuple[str, str, str], filepath_errmsg: str
+    ) -> None:
+        """Validate axis names tuple against known combinations.
+
+        Args:
+            axis_names: Tuple of axis names to validate
+            filepath_errmsg: Error context for meaningful messages
+        """
+
+        if (
+            len(axis_names) != 3
+            or axis_names[0] == ""
+            or axis_names[1] == ""
+            or axis_names[2] == ""
```

Comment:
```suggestion
            if len(axis_names) != 3 or any(name == "" for unit in axis_names)
```

You have this more Pythonic way in the method below this one